src/Controller/Controller_Actions.php

```diff
@@ -151,10 +151,16 @@ public function get_routes() {
 	 */
 	public function route_notices() {
 
+		global $pagenow;
 		/* Prevent actions being displayed on our welcome pages */
-		if ( ! is_admin() ||
-			 ( rgget( 'page' ) === 'gfpdf-getting-started' ) || ( rgget( 'page' ) === 'gfpdf-update' ) || ( defined( 'DOING_AJAX' ) && DOING_AJAX )
-		) {
+		$register_routes = false;
+		$is_gfpdf_page   = \GPDFAPI::get_misc_class()->is_gfpdf_page();
+
+		if ( ! wp_doing_ajax() && is_admin() && ( $pagenow === 'plugins.php' || $is_gfpdf_page || \GFForms::is_gravity_page() ) ) {
```

Comment:
As I mentioned, assign all conditions to individual variables so this IF statement is easy to understad.